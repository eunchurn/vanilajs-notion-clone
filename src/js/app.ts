import { $$ } from "./selector";

interface Content {
  className: string;
  title: string;
  content: string;
}
export function init(content: Content[]) {
  // First content
  const [first] = content;
  const headingArr = $$(first.className, document);
  const heading = document.querySelectorAll(first.className);
  console.log("Selector Array:", headingArr, "Selector NodeList", heading);
  // Second content
  const [, second] = content;
  const section1 = document.querySelector(second.className);
  if (heading) {
    heading[0].innerHTML = first.title;
  }
  if (section1) {
    const parag = document.createElement("p");
    parag.innerHTML = second.content;
    section1.appendChild(parag);
  }
  // Canvas
  const [, , third] = content;
  const canvas = (document.createElement("canvas") as HTMLCanvasElement)
  Object.assign(canvas, { width: 450, height: 600 });
  if (section1) {
    section1.appendChild(canvas);
  }
  const img = new Image();
  img.src = third.content;
  const context = canvas.getContext("2d");
  if (context) {
    img.onload = function () {
      context.drawImage(img, 0, 0, 450, 600);
      console.log("drawn");
    };
  }
};
