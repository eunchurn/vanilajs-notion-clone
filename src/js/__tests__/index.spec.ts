import { JSDOM } from "jsdom";
import { init } from "../app";

const { window } = new JSDOM("<!doctype html><html><body></body></html>");
// Save these two objects in the global space so that libraries/tests
// can hook into them, using the above doc definition.
global.document = window.document;
// global.window = window as typeof globalThis;
global.Image = window.Image;

describe("app", () => {
  beforeEach(() => {
    window.document.body.innerHTML = `
    <header class="heading"></header>
    <section class="section-1"></section>
    <script src="./bundle.js"></script>`;
  });
  it("should render without error", () => {
    init([
      { className: ".heading", title: "Heading", content: "Simple VanillaJS Yypescript Gulp Project Boilerplate" },
      {
        className: ".section-1",
        title: "Section",
        content:
          "Lorem ipsum dolor sit amet consectetur adipisicing elit. Obcaecati ex alias tempore recusandae maxime, illo eum cumque reprehenderit minima molestias qui sequi iste consectetur dolores ducimus ab neque, provident possimus.",
      },
      {
        className: ".cat-canvas",
        title: "Beckhi",
        content: "./public/images/beckhi.jpg",
      },
    ]);
  });
  it("should match snapshot", () => {
    init([
      { className: ".heading", title: "Heading", content: "Simple VanillaJS Yypescript Gulp Project Boilerplate" },
      {
        className: ".section-1",
        title: "Section",
        content:
          "Lorem ipsum dolor sit amet consectetur adipisicing elit. Obcaecati ex alias tempore recusandae maxime, illo eum cumque reprehenderit minima molestias qui sequi iste consectetur dolores ducimus ab neque, provident possimus.",
      },
      {
        className: ".cat-canvas",
        title: "Beckhi",
        content: "./public/images/beckhi.jpg",
      },
    ]);
    expect(window.document.body).toMatchSnapshot();
  });
});
