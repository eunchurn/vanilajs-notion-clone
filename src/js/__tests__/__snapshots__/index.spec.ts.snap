// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should match snapshot 1`] = `
<body>
  
    
  <header
    class="heading"
  >
    Heading
  </header>
  
    
  <section
    class="section-1"
  >
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Obcaecati ex alias tempore recusandae maxime, illo eum cumque reprehenderit minima molestias qui sequi iste consectetur dolores ducimus ab neque, provident possimus.
    </p>
    <canvas
      height="600"
      width="450"
    />
  </section>
  
    
  <script
    src="./bundle.js"
  />
</body>
`;
